*** map.html
--- map.html
***************
*** 1 ****
!     <title>HK Service Map</title>
--- 1 ----
!     <title>HK Service Map Demo</title>
***************
*** 2 ****
--- 2,19 ----
+     <script>
+       function resetDemo() {
+         sessionStorage.removeItem(
+           'file:hk-service-map,/../scripts/unitinfo.js');
+         const returnPath = sessionStorage.getItem('var:,returnPath');
+         window.location.href = returnPath === null ?
+             '.' : window.location.host + returnPath;
+       }
+       const sessionInfo = sessionStorage
+           .getItem('file:hk-service-map,/../scripts/unitinfo.js');
+       if (sessionInfo !== null) {
+         window.longlat = JSON.parse(
+             /^var longlat = (.*?);$/m.exec(sessionInfo)[1]);
+         window.unitinfo = JSON.parse(
+             /^var unitinfo = (.*?);$/m.exec(sessionInfo)[1]);
+       }
+     </script>
    </head>
***************
*** 2,3 ****
--- 2,8 ----
+     <div style="position: fixed; right: .5em; bottom: 2em; z-index: 1000;">
+       <button onclick="resetDemo()" style="padding: .5em; background-color: red;">
+         Reset demo
+       </button>
+     </div>
    </body>
  </html>
