*** map.html
--- map.html
***************
*** 1 ****
!     <title>HK Service Map</title>
--- 1 ----
!     <title>HK Service Map Demo</title>
***************
*** 2 ****
--- 2,37 ----
+     <script>
+       function changeService() {
+         if (window.location.pathname === '/hk-service-map/openlayers.html') {
+           sessionStorage.setItem('var:hk-service-map,useLeaflet', true);
+           window.location.href = 'leaflet.html';
+           return;
+         }
+         if (window.location.pathname === '/hk-service-map/leaflet.html') {
+           sessionStorage.setItem('var:hk-service-map,useLeaflet', false);
+           window.location.href = 'openlayers.html';
+           return;
+         }
+         alert('Not avaiable here, try again on the main map page.');
+       }
+       function resetDemo() {
+         const returnPath = sessionStorage.getItem('var:,returnPath');
+         if (returnPath === null) {
+           sessionStorage.removeItem(
+               'file:hk-service-map,/../scripts/unitinfo.js');
+           window.location.href = '.';
+           return;
+         }
+         sessionStorage.removeItem('var:,returnPath');
+         window.location.href = window.location.protocol + '//' +
+             window.location.host + returnPath;
+       }
+       const sessionInfo = sessionStorage
+           .getItem('file:hk-service-map,/../scripts/unitinfo.js');
+       if (sessionInfo !== null) {
+         window.longlat = JSON.parse(
+             /^var longlat = (.*?);$/m.exec(sessionInfo)[1]);
+         window.unitinfo = JSON.parse(
+             /^var unitinfo = (.*?);$/m.exec(sessionInfo)[1]);
+       }
+     </script>
    </head>
***************
*** 2 ****
--- 2,10 ----
+     <div style="position: fixed; right: .5em; bottom: 2em; z-index: 1000;">
+       <button onclick="changeService()" style="padding: .5em; background-color: teal;">
+         Change Layering Service
+       </button>
+       <button onclick="resetDemo()" style="padding: .5em; background-color: red;">
+         Reset demo
+       </button>
+     </div>
    </head>
